lib/screens/v2/candidate_generator/elevator/closures.ex

```diff
@@ -169,4 +176,15 @@ defmodule Screens.V2.CandidateGenerator.Elevator.Closures do
       ie -> if InformedEntity.parent_station?(ie), do: ie.stop
     end)
   end
+
+  defp has_nearby_redundancy?(%Alert{
+         informed_entities: [%{facility: %{id: informed_facility_id}} | _]
```

Comment:
Here, we check for an affected elevator by matching on just the first informed entity in the alert (and actually, if this happens to not reference a facility, we'll crash ⚠️). But later in `alert_to_elevator_closure` we search through the informed entities for the first one that has a non-null facility. I wonder if we could move things around such that this logic only happens in one place and is therefore consistent — maybe first mapping the alerts to "closures" using `alert_to_elevator_closure`, then rejecting those with nearby redundancy, then constructing the route pills and `Station`s from those that remain.

Reply:
Part of `relevant_alert?/1` is ensuring that all `informed_entities` include a facility and that the facility is the same for all. So we can be sure that any item in the list will include a facility in this case.

That said, I think there is room to clean up some of these helper functions. Will take a look.

Reply:
Ah, gotcha. So there is technically no possibility of a crash here, it's just not super obvious.

Reply:
I think maybe the best way to make that more clear is to move this function call to `relevant_alert?` instead? There's no reason it can be done right after the facility check there.